src/dataclass_binder/_impl.py

```diff
@@ -49,7 +61,7 @@ def _collect_type(field_type: type, context: str) -> type | Binder[Any]:
             return object
         elif not isinstance(field_type, type):
             raise TypeError(f"Annotation for field '{context}' is not a type")
-        elif issubclass(field_type, str | int | float | date | time | timedelta | ModuleType | Path):
+        elif issubclass(field_type, str | int | float | date | time | timedelta | ModuleType | Path | Enum | ReprEnum):
```

Comment:
```suggestion
        elif issubclass(field_type, str | int | float | date | time | timedelta | ModuleType | Path | Enum):
```
As `ReprEnum` is a subclass of `Enum`, we don't need to check for it explicitly, like we don't check for `bool` because it is a subclass of `int`.